src/js/actions/libraries.js

```diff
@@ -0,0 +1,272 @@
+     * different actions and handlers.
+     *
+     * @param {AdobeLibraryElement} element
+     *
+     * @return {Promise}
+     */
+    var createLayerFromElement = function (element) {
+        var appStore = this.flux.store("application"),
+            libStore = this.flux.store("library"),
+            currentDocument = appStore.getCurrentDocument(),
+            currentLibrary = libStore.getCurrentLibrary();
+
+        if (!currentDocument || !currentLibrary) {
+            return Promise.resolve();
+        }
+
+        var docRef = docAdapter.referenceBy.id(currentDocument.id),
+            location = { x: 100, y: 100 },
+            representation = element.getPrimaryRepresentation();
+
+        return Promise.fromNode(function (cb) {
+            representation.getContentPath(cb);
+        }).then(function (path) {
+            var placeObj = libraryAdapter.placeElement(docRef, element, path, location);
+
+            return descriptor.playObject(placeObj);
+        });
+    };
+    createLayerFromElement.reads = [locks.JS_LIBRARIES, locks.JS_DOC];
+    createLayerFromElement.writes = [locks.JS_LIBRARIES, locks.JS_DOC];
+
+    /**
+     * Marks the given library ID as the active one
+     *
+     * @param {string} id
+     *
+     * @return {Promise}
+     */
+    var selectLibrary = function (id) {
+        return this.dispatchAsync(events.libraries.LIBRARY_SELECTED, { id: id });
+    };
+    selectLibrary.reads = [];
+    selectLibrary.writes = [locks.JS_LIBRARIES];
+
+    /**
+     * Creates a new library with the given name
+     *
+     * @param {string} name
+     * @return {Promise.<Library>} Resolves to the created library
+     */
+    var createLibrary = function (name) {
+        var libStore = this.flux.store("library"),
+            libraryCollection = libStore.getLibraryCollection(),
+            newLibrary = libraryCollection.createLibrary(name);
+
+        return this.dispatchAsync(events.libraries.LIBRARY_CREATED, { library: newLibrary })
+            .then(function () {
+                return newLibrary;
+            });
+    };
+    createLibrary.reads = [];
+    createLibrary.writes = [locks.CC_LIBRARIES, locks.JS_LIBRARIES];
+
+    /** 
+     * Removes the current library from the collection
+     *
+     * @return {Promise}
+     */
+    var removeCurrentLibrary = function () {
+        var libStore = this.flux.store("library"),
+            libraryCollection = libStore.getLibraryCollection(),
+            currentLibrary = libStore.getCurrentLibrary();
+
+        if (!libraryCollection || !currentLibrary) {
+            return Promise.resolve();
+        }
+
+        var payload = {
+            id: currentLibrary.id
+        };
+
+        return Promise.fromNode(function (cb) {
+            libraryCollection.removeLibrary(currentLibrary, cb);
+        }).bind(this).then(function () {
+            this.dispatchAsync(events.libraries.LIBRARY_REMOVED, payload);
```

Comment:
Is there a particular reason I need to return an internal promise result, despite having the return in the first promise? Do we care about the result of dispatch?